### DISCUSSION TITLE
drainage and imbibition

### DISCUSSION THREAD

**Discussion by sinaainesaz** [PMEAL/OpenPNM#2855] [Q&A]
Hi,

I was wondering when we are using invasion percolation to predict the relative permeability using water as wetting phase and air as the non-wetting phase.
like this:
# Relative permeability
        ip = op.algorithms.InvasionPercolation(network=pn, phase=oil)
        Finlets_init = pn.pores('left')
        Finlets=([Finlets_init[x] for x in range(0, len(Finlets_init), 2)])
        ip.set_inlet_BC(pores=Finlets)
        ip.run()
        
    
        
        flow_in = pn.pores('left')
        flow_out = pn.pores('right')

        model_mp_cond = op.models.physics.multiphase.conduit_conductance
        water.add_model(model=model_mp_cond, propname='throat.conduit_hydraulic_conductance',
                      throat_conductance='throat.hydraulic_conductance', mode='medium', regen_mode='deferred')
        oil.add_model(model=model_mp_cond, propname='throat.conduit_hydraulic_conductance',
                      throat_conductance='throat.hydraulic_conductance', mode='medium', regen_mode='deferred')

1. Does it simulates the Primary drainage process?
2. And if we wanna simulate the imbibition, we should do the process above and after calculation without cutting the code, we only change the flow_in to right and flow out to left and in invasion percolation put the non-wetting phase as water?

**Comment by likeduck** [comment]
According to my current understanding, OpenPNM's two-phase flow does not contain primary drainage, and there is no adsorption layer. But we should know that this is exactly what we need to do in the mathematical model.
